src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/haflows/HaFlowDiversitySpec.groovy

```diff
@@ -149,16 +159,17 @@ class HaFlowDiversitySpec extends HealthCheckSpecification {
         assumeTrue(swT != null, "Unable to find suitable switches")
 
         when: "Create an HA-Flow without diversity"
-        def haFlow1 = HaFlowExtended.build(swT, northboundV2, topology).create()
+        def haFlow1 = haFlowFactory.getRandom(swT)
 
         and: "Create a Y-Flow diverse with previously created HA-Flow"
         def yFlowRequest1 = yFlowHelper.randomYFlow(swT, false)
                 .tap { it.diverseFlowId = haFlow1.haFlowId }
         def yFlow = yFlowHelper.addYFlow(yFlowRequest1)
 
         and: "Create an additional HA-Flow diverse with Y-Flow that has another HA-Flow in diverse group"
-        def haFlow2 = HaFlowExtended.build(swT, northboundV2, topology, false, haFlow1.occupiedEndpoints())
-                .withDiverseFlow(yFlow.getYFlowId()).create()
+        def haFlow2 = haFlowFactory.getBuilder(swT, false, haFlow1.occupiedEndpoints())
```

Comment:
If you won't make `.build()` or `.create()` method to wait for status, can we at least chain the flow creation and waiting for status? E.g.
```
def haFlow2 = haFlowFactory.getBuilder()...
                                                    .buld())
                          .waitForBeingInState()..
```

Reply:
Done.